### DISCUSSION TITLE
Is it possible to add a system notification for Window with an application icon?

### DISCUSSION THREAD

**Discussion by lujihong** [wailsapp/wails#3349] [Q&A]
### Describe the solution you'd like

Is it possible to add a system notification for Window with an application icon?

**Comment by lujihong** [comment]
**I made an implementation with the following code, but it doesn't always feel very good**

`package win

import (
	"AILaunchMaster/pkg/logger"
	"github.com/lxn/walk"
)

var (
	mw   *walk.MainWindow
	ni   *walk.NotifyIcon
	icon *walk.Icon
)

func createMainWindow() error {
	var err error

	// 创建一个 MainWindow 对象
	mw, err = walk.NewMainWindow()
	if err != nil {
		return err
	}

	go func() {
		mw.Run()
	}()

	return nil
}

func createNotifyIcon() error {
	var err error

	// 从文件中加载图标
	icon, err = walk.Resources.Icon("pkg/icons/icon.ico")
	if err != nil {
		return err
	}

	// 创建 NotifyIcon，并在退出时清理它
	ni, err = walk.NewNotifyIcon(mw)
	if err != nil {
		return err
	}

	// 设置图标
	if err := ni.SetIcon(icon); err != nil {
		return err
	}

	// 设置为不可见
	if err := ni.SetVisible(false); err != nil {
		return err
	}

	return nil
}

func init() {
	if err := createMainWindow(); err != nil {
		logger.LogError(err.Error())
	}

	if err := createNotifyIcon(); err != nil {
		logger.LogError(err.Error())
	}
}

// ShowMessage 显示消息通知
func ShowMessage(title, msg string) error {
	err := ni.ShowMessage(title, msg)
	if err != nil {
		return err
	}
	return nil
}

// ShowInfo 显示信息通知
func ShowInfo(title, msg string) error {
	err := ni.ShowInfo(title, msg)
	if err != nil {
		return err
	}
	return nil
}

// ShowError 显示错误通知
func ShowError(title, msg string) error {
	err := ni.ShowError(title, msg)
	if err != nil {
		return err
	}
	return nil
}

// ShowWarning 显示警告通知
func ShowWarning(title, msg string) error {
	err := ni.ShowWarning(title, msg)
	if err != nil {
		return err
	}
	return nil
}
`

**Comment by fyonecon** [comment]
走错框架了吧你，，，win里面发送通知直接调用webview2里面的web通知就行了。。。
